src/bin/bacs_system_build_mono_develop.py

```diff
@@ -80,13 +80,16 @@ def create_executable(output, executable, relative=False):
     executable = shlex.quote(executable)
     if relative:
         executable = '"$(dirname "$0")/"' + executable
-    with open(executable, 'a') as exe:
-        os.fchmod(exe.fileno(), os.fstat(exe.fileno()).st_mode | 0o111)
     with open(output, 'w') as out:
         print('#!/bin/sh -e', file=out)
         print('exec', executable, '"$@"', file=out)
         os.fchmod(out.fileno(), os.fstat(out.fileno()).st_mode | 0o111)
 
+def set_execute_bit(binary):
+    binary = shlex.quote(binary)
```

Comment:
This is incorrect and will not work. You copied it from the code which formatted binary name for shell usage, however here you use it for syscall. In case binary contains spaces and has actually to be quoted it will fail. Use binary directly, don't quote it.